internal/bminventory/inventory.go

```diff
@@ -1435,9 +1458,11 @@ func (b *bareMetalInventory) InstallSingleDay2HostInternal(ctx context.Context,
 	if h, err = b.getHost(ctx, infraEnvId.String(), hostId.String()); err != nil {
 		return err
 	}
+
 	// auto select host roles if not selected yet.
 	err = b.db.Transaction(func(tx *gorm.DB) error {
-		if _, err = b.hostApi.AutoAssignRole(ctx, &h.Host, tx); err != nil {
+		// no need to specify expected master count for day2 hosts, as their suggested role will always be worker
```

Comment:
In day2, assisted doesn't always have a way to know the roles of the other hosts from day1. Therefore, the usage of the new `controlPlaneCount` is problematic. I believe for this scenario it would be best to set the role manually in case of master